File: applications/luci-statistics/luasrc/model/cbi/luci_statistics/tcpconns.lua
--[[

Luci configuration model for statistics - collectd tcpconns plugin configuration
(c) 2008 Freifunk Leipzig / Jo-Philipp Wich <xm@leipzig.freifunk.net>

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

        http://www.apache.org/licenses/LICENSE-2.0

$Id$

]]--

m = Map("luci_statistics",
	translate("TCPConns Plugin Configuration"),
	translate(
		"The tcpconns plugin collects informations about open tcp " ..
		"connections on selected ports."
	))

-- collectd_tcpconns config section
s = m:section( NamedSection, "collectd_tcpconns", "luci_statistics" )

-- collectd_tcpconns.enable
enable = s:option( Flag, "enable", translate("Enable this plugin") )
enable.default = 0

-- collectd_tcpconns.listeningports (ListeningPorts)
listeningports = s:option( Flag, "ListeningPorts", translate("Monitor all local listen ports") )
listeningports.default = 1
listeningports:depends( "enable", 1 )

-- collectd_tcpconns.localports (LocalPort)
localports = s:option( Value, "LocalPorts", translate("Monitor local ports") )
localports.optional = true
localports:depends( "enable", 1 )

-- collectd_tcpconns.remoteports (RemotePort)
remoteports = s:option( Value, "RemotePorts", translate("Monitor remote ports") )
remoteports.optional = true
remoteports:depends( "enable", 1 )

return m

File: applications/luci-app-statistics/htdocs/luci-static/resources/view/statistics/plugins/tcpconns.js
'use strict';
'require baseclass';
'require form';

return baseclass.extend({
	title: _('TCPConns Plugin Configuration'),
	description: _('The tcpconns plugin collects information about open tcp connections on selected ports.'),

	addFormOptions: function(s) {
		var o;

		o = s.option(form.Flag, 'enable', _('Enable this plugin'));

		o = s.option(form.Flag, 'ListeningPorts', _('Monitor all local listen ports'));
		o.depends('enable', '1');
		o.rmempty = false;
		o.default = '0';

		o = s.option(form.DynamicList, 'LocalPorts', _('Monitor local ports'));
		o.optional = true;
		o.datatype = 'port';
		o.default = '22';
		o.depends('enable', '1');

		o = s.option(form.DynamicList, 'RemotePorts', _('Monitor remote ports'));
		o.optional = true;
		o.datatype = 'port';
		o.depends('enable', '1');

		o = s.option(form.Flag, 'AllPortsSummary', _('Summary of all ports'));
		o.rmempty = false;
		o.depends('enable', '1');
	},

	configSummary: function(section) {
		var lports = L.toArray(section.LocalPorts),
		    rports = L.toArray(section.RemotePorts),
		    listen = section.ListeningPorts == '1',
		    summary = section.AllPortsSummary == '1';

		return _('Monitoring %s and %s, %s %s').format(
			N_(lports.length, 'one local', '%d local').format(lports.length),
			N_(rports.length, 'one remote port', '%d remote ports').format(rports.length),
			listen ? _('all local listening ports,') : '',
			summary ? _('summary of all ports') : _('no summary')
		);
	}
});
